### PR TITLE
[Caffe2] Always build local protobuf library with -fPIC

### PR DESCRIPTION
**Pull request by bddppq**
@Yangqing @pjh5

### CODE REVIEW

**Comment by bddppq** [comment]
@pietern @bwasti

**Review by pietern — APPROVED** [review]

**Review by pietern — APPROVED** [review]

**Review by pietern — APPROVED** [review]

File: cmake/ProtoBuf.cmake [hunk 1]
```diff
@@ -41,10 +38,14 @@ macro(custom_protobuf_find)
     endif()
   endif()
 
+  set(__caffe2_CMAKE_POSITION_INDEPENDENT_CODE ${CMAKE_POSITION_INDEPENDENT_CODE})
```
Comment (pjh5) [inline cmake/ProtoBuf.cmake]: If CAFFE2_LINK_LOCAL_PROTOBUF is ON, then won't this line override the original CMAKE_POSITION_INDEPENDENT_CODE that was saved into __caffe2_CMAKE_pic ?

Comment (bddppq) [reply cmake/ProtoBuf.cmake]: I guess you are saying __caffe2_CMAKE_POSITION_INDEPENDENT_CODE, these variables are meant to be tmp.

Comment (pjh5) [reply cmake/ProtoBuf.cmake]: if CAFFE2_LINK_LOCAL_PROTOBUF is ON then line 33 will overwrite CMAKE_POSITION_INDEPENDENT_CODE to ON, and then line 41 will always set __caffe2_CMAKE_POSITION_INDEPENDENT_CODE to ON, then line 46 will always set CMAKE_POSITION_INDEPENDENT_CODE back to ON

Comment (bddppq) [reply cmake/ProtoBuf.cmake]: discussed with @pjh5 offline, there is no issue.

**Pull request merged**